File: Baekjoon/Bronze/1159_농구_게임/1159.py
"""
- 분류 : 구현 문자열열
- 성능 요약
    - 메모리 : 32412 KB
    - 시간 : 36 ms
- 제출 일자 : 2025-01-23 12:57
"""

N = int(input())

member = []
for i in range(N):
    member.append(input())

member.sort()

count = 1
first = ""
result = ""

for idx in range(N):
    if idx == 0:
        first = member[idx][0]
    else:
        if first == member[idx][0]:
            count += 1
        else:
            if count >= 5:
                result += first
            count = 1
            first = member[idx][0]

if count >= 5:
    result += first

if result == "":
    print("PREDAJA")
else:
    print(result)

File: Baekjoon/Bronze/1159_농구_게임/1159.js
`
- 분류 : 구현, 문자열
- 성능 요약
    - 메모리 : 9340 KB
    - 시간 : 92 ms
- 제출 일자 : 2025-02-24 08:58
`

const fs = require("fs"); 
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");
// const input = fs.readFileSync("input.txt").toString().trim().split("\n");

const N = parseInt(input[0])
const count = {};  // 각 성의 첫 글자 빈도를 저장할 객체

for (let i = 1; i <= N; i++) {
    const firstLetter = input[i][0]; // 성의 첫 글자만 가져오기
    count[firstLetter] = (count[firstLetter] || 0) + 1;
}

// 5명 이상인 첫 글자만 필터링 후 정렬
const result = Object.keys(count)
    .filter(letter => count[letter] >= 5)
    .sort()
    .join("");  // 문자열로 변환

console.log(result.length > 0 ? result : "PREDAJA");